messages/event_subscription.go

```diff
@@ -0,0 +1,90 @@
+package messages
+
+import (
+	"github.com/0xPolygon/go-ibft/messages/proto"
+)
+
+type eventSubscription struct {
+	// details contains the details of the event subscription
+	details SubscriptionDetails
+
+	// outputCh is the update channel for the subscriber
+	outputCh chan uint64
+
+	// doneCh is the channel for handling stop signals
+	doneCh chan struct{}
+
+	// notifyCh is the channel for receiving event requests
+	notifyCh chan uint64
+}
+
+// close stops the event subscription
+func (es *eventSubscription) close() {
+	close(es.doneCh)
+}
+
+// runLoop is the main loop that listens for notifications and handles the event / close signals
+func (es *eventSubscription) runLoop() {
+	defer close(es.outputCh)
+
+	for {
+		select {
+		case <-es.doneCh: // Break if a close signal has been received
+			return
+		case round := <-es.notifyCh: // Listen for new events to appear
+			select {
+			case <-es.doneCh: // Break if a close signal has been received
+				return
+			case es.outputCh <- round: // Pass the event to the output
+			}
+		}
+	}
+}
+
+// eventSupported checks if any notification event needs to be triggered
+func (es *eventSubscription) eventSupported(
+	messageType proto.MessageType,
+	view *proto.View,
+	totalMessages int,
+) bool {
+	// The heights must match
+	if view.Height != es.details.View.Height {
+		return false
+	}
+
+	// The round can be treated as a min round
+	if es.details.HasMinRound &&
+		view.Round >= es.details.View.Round {
+		return false
+	}
+
+	// The rounds must match
+	if view.Round != es.details.View.Round {
```

Comment:
Needs to match even if `es.details.HasMinRoun` is true? What is previous comparison?

Reply:
If the min round is set, then we evaluate the higher round as valid.

But if `HasMInRound` is false, then we want the exact round

https://github.com/0xPolygon/go-ibft/blob/2dbfb199c35d0fb2b4cbf680d5385639fd2b2585/messages/event_subscription.go#L56-L59

Reply:
No no no, it's not the answer.

Let's say `es.details.HasMinRound` is `true` and `es.details.View.Round` is 5. If view.round is 6, first comparison passes (`es.details.HasMinRound && view.Round >= es.details.View.Round`) but the second catches (`view.Round != es.details.View.Round`). So the result is `false` regardless of that you set `es.details.HasMinRound` or not.

@zivkovicmilos 

Am I correct?

Reply:
You are absolutely correct, only thing that I would add is that the first conditional would return `false` instead of going forward 🤦‍♂️ 

I've fixed this, as it's not the desired behavior we envisioned (added also a test to cover these cases):

[9e1ce4b](https://github.com/0xPolygon/go-ibft/pull/21/commits/9e1ce4bd1b6884605c3421a72160729024c27487)